pallets/runtime/tests/src/sto_test.rs

```diff
@@ -0,0 +1,332 @@
+
+    // Register a venue
+    let venue_counter = Settlement::venue_counter();
+    assert_ok!(Settlement::create_venue(
+        alice_signed.clone(),
+        VenueDetails::default(),
+        vec![alice],
+        VenueType::Sto
+    ));
+
+    let amount = 100u128;
+    let alice_init_offering = Asset::balance_of(&offering_ticker, alice_did);
+    let bob_init_offering = Asset::balance_of(&offering_ticker, bob_did);
+    let alice_init_raise = Asset::balance_of(&raise_ticker, alice_did);
+    let bob_init_raise = Asset::balance_of(&raise_ticker, bob_did);
+
+    // Alice starts a fundraiser
+    let fundraiser_id = STO::fundraiser_count(offering_ticker);
+    assert_ok!(STO::create_fundraiser(
+        alice_signed.clone(),
+        alice_portfolio,
+        offering_ticker,
+        alice_portfolio,
+        raise_ticker,
+        vec![PriceTier {
+            total: 1_000_000u128,
+            price: 1u128
+        }],
+        venue_counter,
+        None,
+        None,
+    ));
+    assert_eq!(
+        STO::fundraisers(offering_ticker, fundraiser_id),
+        Some(Fundraiser {
+            creator: alice_did,
+            offering_portfolio: alice_portfolio,
+            offering_asset: offering_ticker,
+            raising_portfolio: alice_portfolio,
+            raising_asset: raise_ticker,
+            tiers: vec![FundraiserTier {
+                total: 1_000_000u128,
+                remaining: 1_000_000u128,
+                price: 1u128
+            }],
+            venue_id: venue_counter,
+            start: Timestamp::get(),
+            end: None,
+            frozen: false
+        })
+    );
+
+    assert_eq!(
+        Asset::balance_of(&offering_ticker, alice_did),
+        alice_init_offering
+    );
+    assert_eq!(
+        Asset::balance_of(&offering_ticker, bob_did),
+        bob_init_offering
+    );
+    assert_eq!(
+        Asset::balance_of(&raise_ticker, alice_did),
+        alice_init_raise
+    );
+    assert_eq!(Asset::balance_of(&raise_ticker, bob_did), bob_init_raise);
+
+    // Bob invests in Alice's fundraiser
+    assert_ok!(STO::invest(
+        bob_signed.clone(),
+        bob_portfolio,
+        bob_portfolio,
+        offering_ticker,
+        fundraiser_id,
+        amount.into(),
+        Some(2u128),
+        None
+    ));
+
+    assert_eq!(
+        STO::fundraisers(offering_ticker, 1),
+        Some(Fundraiser {
+            creator: alice_did,
+            offering_portfolio: alice_portfolio,
+            offering_asset: offering_ticker,
+            raising_portfolio: alice_portfolio,
+            raising_asset: raise_ticker,
+            tiers: vec![FundraiserTier {
+                total: 1_000_000u128,
+                remaining: (1_000_000 - amount).into(),
+                price: 1u128
+            }],
+            venue_id: venue_counter,
+            start: Timestamp::get(),
+            end: None,
+            frozen: false
+        })
+    );
```

Comment:
This is almost the same as the assert before, aside from `- amount` and the id; let's dedup.